syntax = "proto3";
package OpenSourceProjects;

message Project {
    message Date {
        int32 year = 1;
        int32 month = 2;
        int32 day = 3;
    }

    string name = 1;     //Proper project name
    string host = 2;     //Hosting/control foundation or company
    Date inception = 3;   //Date project was open sourced
}

message CreateResult {
    int32 code = 1;
    string message = 2;
}

message GetArg {
  string name = 1;
}

service Projects {
    rpc Get(GetArg) returns (Project);
    rpc Create(Project) returns (CreateResult);
}
